apps/storybook/src/Selection.stories.tsx

```diff
@@ -24,6 +30,45 @@ function vectorToStr(vec: Vector2) {
   return `(${format('.2f')(vec.x)}, ${format('.2f')(vec.y)})`;
 }
 
+const X = range(-5, 5, 0.01);
+const Y = X.map((x) => Math.exp((-x * x) / 0.2));
+
+function GaussianCurve() {
+  return <DataCurve abscissas={X} ordinates={Y} color="blue" />;
+}
+
+const modifierKeyArgType = {
+  control: { type: 'inline-radio' },
+  options: ['Alt', 'Control', 'Shift'],
+};
+
+const SelectionArgType = {
+  control: { type: 'inline-radio' },
+  options: ['line', 'rectangle'],
+};
+
+function DataSelectionMesh(props: SelectionMeshProps) {
+  const { onSelection, modifierKey, children } = props;
+  const { worldToData } = useAxisSystemContext();
+
+  return (
+    <SelectionMesh
+      onSelection={
+        onSelection
+          ? ({ startPoint, endPoint }) =>
+              onSelection({
+                startPoint: worldToData(startPoint),
+                endPoint: worldToData(endPoint),
+              })
+          : undefined
+      }
+      modifierKey={modifierKey}
+    >
+      {children}
+    </SelectionMesh>
+  );
+}
```

Comment:
Due to https://github.com/silx-kit/h5web/pull/981#discussion_r813019807, `onSelection` parameters are now in world coordinates.

That means that for the story that prints the selection points in data coordinates, we need for a component that has access to `AxisSystemContext` to call `worldToData`.

Perhaps it is worth having this component in `@h5web/lib` ?

Reply:
Would it make sense for `SelectionLine` and `SelectionRect` to also have a prop `withWorldCoordinates` so that we wouldn't have to call `dataToWorld` here? This would allow inlining this component in the story template.

For now, though, you could maybe just remove the `useWorldCoordinates` argument and inline this component in the template right away.

Reply:
We could.

But then, we must be aware that we cannot use components with `withWorldCoordinates` to persist selections at the risk of falling back on the issue solved by https://github.com/silx-kit/h5web/pull/935

Reply:
`SelectionLine` and `SelectionRect` now consider that props are in data coordinates to avoid any resizing problems